holoviews/core/data/grid.py

```diff
@@ -243,7 +243,7 @@ def canonicalize(cls, dataset, data, data_coords=None, virtual_coords=[]):
         dropped = [dims.index(d) for d in dims
                    if d not in dataset.kdims+virtual_coords]
         if dropped:
-            data = data.squeeze(axis=tuple(dropped))
+            data = np.squeeze(data, axis=tuple(dropped))
```

Comment:
So calling `np.squeeze` will probably make this non-lazy. Not sure if that matters in this case. Just a friendly note.

Reply:
NVM. Turns out I'm wrong. `np.squeeze` works just fine on Dask Arrays. Sorry for the noise.

Reply:
It seems to work okay in my simple test case at least, at least it's still a dask array after ``np.squeeze``. Could it still be non-lazy? Here's an example:

```python
darr = da.from_array(np.random.rand(30, 20, 20), (3, 2, 2))
ds = hv.Dataset((range(20), range(20), range(30), darr),
                ['x', 'y', 'z'], 'Value')
z_stack = ds.to(hv.Image, ['x', 'y'])
z_stack.last.data
```

```
{'Value': dask.array<squeeze, shape=(20, 20), dtype=float64, chunksize=(20, 20)>,
 'x': array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
        17, 18, 19]),
 'y': array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
        17, 18, 19]),
 'z': array([29])}
```

Reply:
Great, thanks! I noticed that since dask 1.7.0 you can use ``np.squeeze`` with an ``axis`` argument but ``da.Array.squeeze`` does not accept ``axis``, is this an oversight?

Reply:
Thanks for pointing that out. Agree it is probably an oversight. Raised as issue ( https://github.com/dask/dask/issues/3259 ). Should be trivial to fix.